pysph/sph/surface_tension.py

```diff
@@ -19,6 +19,198 @@
 from math import sqrt
 
 
+class SurfaceForceAdami(Equation):
+    def initialize(self, d_au, d_av, d_idx):
+        d_au[d_idx] = 0.0
+        d_av[d_idx] = 0.0
+
+    def loop(self, d_au, d_av, d_idx, d_m, DWIJ, d_pi00, d_pi01, d_pi10,
+             d_pi11, s_pi00, s_pi01, s_pi10, s_pi11, d_V, s_V, s_idx):
+        s2 = s_V[s_idx]*s_V[s_idx]
+        f1 = (d_pi00[d_idx]/(d_V[d_idx]*d_V[d_idx])+s_pi00[s_idx]/s2)
+        f2 = (d_pi01[d_idx]/(d_V[d_idx]*d_V[d_idx])+s_pi01[s_idx]/s2)
+        f3 = (d_pi10[d_idx]/(d_V[d_idx]*d_V[d_idx])+s_pi10[s_idx]/s2)
+        f4 = (d_pi11[d_idx]/(d_V[d_idx]*d_V[d_idx])+s_pi11[s_idx]/s2)
+        d_au[d_idx] += (DWIJ[0]*f1 + DWIJ[1]*f2)/d_m[d_idx]
+        d_av[d_idx] += (DWIJ[0]*f3 + DWIJ[1]*f4)/d_m[d_idx]
```

Comment:
So this will work only in 2D, can you fix this to work in 3D also?

Reply:
Sir,

I will need to formulate the whole matrix and rewrite the formulas. I'll do it when I get some free time for sure.

Reply:
Perhaps it is easier to just setup a matrix and do a matrix multiply, there are some simple routines that will do this for you in `pysph.sph.wc.linalg`.

Reply:
This has been fixed to work in 3D also now. Please take a look